### ISSUE TITLE
`NotJavadoc` pattern does not allow Javadoc on module declarations

### ISSUE THREAD

**Issue by codeleverage** [google/error-prone#3923]
The `NotJavadoc` pattern should allow Javadoc on module declarations.  

```java
/** Documentation. */
module test.foo {
  exports test.foo;
}
```

For the above `module-info.java`, I expect no warnings/errors but get:
```
[WARNING] ../src/main/java/module-info.java:[1,1] [NotJavadoc] Avoid using /** for comments which aren't actually Javadoc.
    (see https://errorprone.info/bugpattern/NotJavadoc)
  Did you mean '/* Documentation. */'?
```
